spec/system/ai_bot/share_spec.rb

```diff
@@ -35,11 +35,56 @@
     bot_user.update!(username: "gpt-4")
 
     Group.refresh_automatic_groups!
-    pm
-    pm_posts
+  end
+
+  it "can share a conversation with a persona user" do
+    clip_text = nil
+
+    persona = Fabricate(:ai_persona, name: "Tester")
+    persona.create_user!
+
+    Fabricate(:post, topic: pm, user: admin, raw: "How do I do stuff?")
+    Fabricate(:post, topic: pm, user: persona.user, raw: "No idea")
+
+    visit(pm.url)
+
+    # clipboard functionality is extremely hard to test
+    # we would need special permissions in chrome driver to enable full access
+    # instead we use a secret variable to signal that we want to store clipboard
+    # data in window.discourseAiClipboard
+    page.execute_script("window.discourseAiTestClipboard = true")
```

Comment:
We already have this sorted in core, nothing fancy needed. We have a special "CDP" page object:


```ruby
let(:cdp) { PageObjects::CDP.new }
```

Which is the chrome dev tools comms protocol. Once you know something is on the clipboard, all you have to do is:

```ruby
expect(cdp.read_clipboard).to eq(clip_text)
```

We have several core and chat examples e.g. https://github.com/discourse/discourse/blob/b057f1b2b4b534c6618e8c19ea32166722a9ee01/spec/system/post_selection_copy_quote_spec.rb . Sometimes you need to use `.chomp` too

Reply:
You can also use this CDP thing to e.g. limit network speed, we have a couple of tests that slow down the network for uploads for example. It can do tons of other stuff too

Reply:
will try cdp, I tried it previously and had no luck, but will give it a shot again.

Reply:
Oh weird, yeah definitely LMK if it's not working and I can check out locally. It's working fine over many tests in core + chat

Reply:
all working now... just pushed a re-implementation.